crates/html_node/src/lib.rs

```diff
@@ -39,6 +39,21 @@ fn init() {
     }
 }
 
+#[napi_derive::napi(object)]
+#[derive(Debug, Serialize)]
+pub struct Diagnostic {
+    pub level: String,
+    pub message: String,
+    pub span: serde_json::Value,
+}
+
+#[napi_derive::napi(object)]
+#[derive(Debug, Serialize)]
+pub struct TransformOutput {
+    pub code: String,
+    pub errors: Vec<Diagnostic>,
```

Comment:
Please add `skip_serializing_if`
That's the reason I asked to remove needless `Option`

Reply:
Do you mean `#[serde(skip_serializing_if = "Vec::is_empty")]`? Because it is still return empty vector

Reply:
Yes, `Option<Vec<T>>` is meaningless only if we are using `skip_serializing_if`

Reply:
But `#[serde(skip_serializing_if = "Vec::is_empty")]` is not working...

Output:
```
{
  code: '<!doctype html><h1>My First Heading</h1>\n' +
    '<p style="color:red">My first paragraph.</p>\n' +
    '\n' +
    '\n' +
    '\n' +
    '<br>\n' +
    '\n' +
    '<style test="color:red">a{color:red}</style>',
  errors: []
}

```

i.e. we still ahve empty array

Reply:
It works only with `Option<...>`

Reply:
Why are empty `errors` required?

Reply:
It is not required, if I use:
```
#[napi_derive::napi(object)]
#[derive(Debug, Serialize)]
pub struct TransformOutput {
    pub code: String,
    #[serde(skip_serializing_if = "Vec::is_empty")]
    pub errors: Vec<Diagnostic>,
}
```

Our output is always:
```
{
  code: '<!doctype html><h1>My First Heading</h1>\n' +
    '<p style="color:red">My first paragraph.</p>\n' +
    '\n' +
    '\n' +
    '\n' +
    '<br>\n' +
    '\n' +
    '<style test="color:red">a{color:red}</style>',
  errors: []
}
``` 

Because it doesn't work, it only works when we have `Option<...>`

Reply:
Ah.. Can you revert it then?
I didn't know that napi does not support it

Reply:
@kdy1 Done :+1: